### ISSUE TITLE
How should I use Evision.Feature2D.detect!/2?

### ISSUE THREAD

**Issue by zacky1972** [cocoa-xu/evision#49]
Hi,

The following python code works well:

```python
import cv2

img = cv2.imread("test_img.jpg")
gray = cv2.cvtColor(img, cv2.COLOR_RGB2GRAY)
sift = cv2.AKAZE_create()
kp = sift.detect(gray)
```

However, the following elixir code doesn't work well:

```elixir
Mix.install([{:evision, "~> 0.1.0-dev", github: "cocoa-xu/evision", branch: "main"}])

img = Evision.imread!("test_img.jpg")
gray = Evision.cvtColor(img, Evision.cv_COLOR_RGB2GRAY)
sift = Evision.AKAZE.create!()
kp = Evision.Feature2D.detect!(sift, gray)
```

I got the following error:

```elixir
** (ArgumentError) argument error
    (evision 0.1.0-dev) :erl_cv_nif.evision_cv_Feature2D_detect(#Reference<0.2183360161.3024486401.45738>, [image: #Reference<0.2183360161.3024486402.45065>])
    (evision 0.1.0-dev) lib/generated/opencv_feature2d.ex:141: Evision.Feature2D.detect!/2
```

I don't understand the `Evision.Feature2D.detect!/2` function because it is not documented enough, so please let me to know how I should use the function.

**Comment by cocoa-xu** [comment]
Hi, I can confirm this is an issue related to inheritance handling. Will fix this in a few minutes! After the fix, the following code should work

```elixir
Mix.install([{:evision, "~> 0.1.0-dev", github: "cocoa-xu/evision", branch: "main"}])

img = Evision.imread!("test_img.jpg")
gray = Evision.cvtColor(img, Evision.cv_COLOR_RGB2GRAY)
sift = Evision.AKAZE.create!()
kp = Evision.AKAZE.detect!(sift, gray)
```

**Comment by zacky1972** [comment]
I tried to update as follows:

```
mix deps.update evision
```

And then, I also tried to update as follows:

```
rm -rf mix.lock deps _build
mix deps.get
```

However, even in both case, my iex can't find `Evision.AKAZE.detect!`

**Comment by cocoa-xu** [comment]
Hi, I'm still waiting for the CI to finish [https://github.com/cocoa-xu/evision/actions/runs/2014817405](https://github.com/cocoa-xu/evision/actions/runs/2014817405). I'll merge it as soon as the CI passed. 😃

**Comment by cocoa-xu** [comment]
Okay, the fix (#50) has been merged. Please feel free to reopen this issue if the PR didn't work. ^_^

**Comment by zacky1972** [comment]
I found it fixed! Thank you.
